### ISSUE TITLE
Light fail after loading world

### ISSUE THREAD

**Issue by AppPhil** [baseball435/TerraLegion#10]
This is how the world looked after creating it and destroying some blocks:
removed cause of old textures

This is how the world looked after loading the world again (after closing the application):
removed cause of old textures

**Comment by jmrapp1** [comment]
Yes this is a known issue, I forgot to add it so thank you for reminding me. When a chunk is first generated it has to calculate the light values. The way that it does that is it goes to each row (left to right) calculating the highest tile in the chunk and then calculating the light value only downwards. Because those two tiles were the highest tiles in that row, it started the light value at its highest amount and then went downwards. It doesn't take into account the height of neighboring tiles which is what needs to be done.

**Comment by AppPhil** [comment]
In this video you can see more light fails after loading:
removed cause of old textures
